### PR TITLE
feat(Runner): set the working directory to the output path

### PR DESCRIPTION
**Pull request by ouuan**
## Description

Set the working directory to the output path when executing programs.

## Motivation and Context

To prevent creating files in random places, if you write to files in the code. Perhaps it's also useful in other situations.

## How Has This Been Tested?

On Arch Linux.

## Checklist

- [x] I have documented these changes in [CHANGELOG.md](https://github.com/cpeditor/cpeditor/blob/master/doc/CHANGELOG.md), or these changes are not notable.

### CODE REVIEW

File: doc/CHANGELOG.md [hunk 1]
```diff
@@ -2,15 +2,25 @@
 
 ### Added
 
-- Open recent files. (#527)
-- Reveal the executable file in file manager. (#537)
+-   Open recent files. (#527)
```
Comment (coder3101) [inline doc/CHANGELOG.md:5]: What's changed here? Unnecessary changes.

File: doc/CHANGELOG.md [hunk 2]
```diff
@@ -2,15 +2,25 @@
...
+-   Reveal the executable file in file manager. (#537)
 
 ### Fixed
 
-- Fixed the diff viewer crashes when the output is empty. (#517)
-- Fixed the background color of the diff viewer is black on Windows with the Dark Fusion theme. (#519 and #520)
-- Fixed the old problem URL is not restored when opening an old file with that option on. (#522)
-- Fixed the performance issue that makes the editor extremely slow when there are many lines. (cpeditor/QCodeEditor#27)
+-   Fixed the diff viewer crashes when the output is empty. (#517)
```
Comment (coder3101) [inline doc/CHANGELOG.md:10]: Same unnecessary changes. This overwrites git blame, a quick way to find which feature in changelog was done by whom.

Comment (ouuan) [reply doc/CHANGELOG.md:10]: It is better to use indents of width 4, more Markdown implementations support nested list with fixed-width indents, especially when the nested list is complicated.

I didn't think of `git blame` because I thought the PR number is much more useful. If you think it's important, I can revert these changes. However, I'm wondering whether the author of the changelog is always who implemented the feature. I usually use `git blame` to find the commit which introduces a bug, instead of to find the author of something.

Comment (ouuan) [reply doc/CHANGELOG.md:10]: In fact, I didn't change any author:

image

Comment (ouuan) [reply doc/CHANGELOG.md:10]: In fact, I didn't change any author:

image

Do you think it's much easier to go to the PRs in the blame page instead of by the PR number in the brackets?

**Comment by coder3101** [comment]
Resolve conflicts and I'll approve.

File: doc/CHANGELOG.md (hunk 2 above)
Comment (coder3101) [reply doc/CHANGELOG.md:11]: It works offline. Since you have not changed anything it doesn't matter.
